rocketpy/mathutils/function.py

```diff
@@ -133,25 +128,39 @@ def set_outputs(self, outputs):
         return self
 
     def set_source(self, source):
-        """Set the source which defines the output of the function giving a
-        certain input.
+        """Sets the data source for the function, defining how the function
+        produces output from a given input.
 
         Parameters
         ----------
-        source : function, scalar, ndarray, string, Function
-            The actual function. If type is function, it will be called for
-            evaluation. If type is int or float, it will be treated as a
-            constant function. If ndarray, its points will be used for
-            interpolation. An ndarray should be as [(x0, y0, z0), (x1, y1, z1),
-            (x2, y2, z2), ...] where x0 and y0 are inputs and z0 is output. If
-            string, imports file named by the string and treats it as csv.
-            The file is converted into ndarray and should not have headers.
-            If the source is a Function, its source will be copied and another
-            Function will be created following the new inputs and outputs.
+        source : function, scalar, ndarray, string, or Function
+            The data source to be used for the function:
+
+            - Python function: Called for evaluation with input values.
```

Comment:
I believe the correct term here would be callable. It does not need to be a Python function. Any callable will do, such as classes with the `__call__` magic method, or any class method. See https://realpython.com/python-callable-instances/#:~:text=In%20Python%2C%20a%20callable%20is,examples%20of%20callables%20in%20Python. for details.

```suggestion
            - Callable: Called for evaluation with input values. Must have the desired inputs as arguments and return a single output value. Input order is important. Example: Python functions, classes, and methods.
```

Reply:
PLease keep in mind thath "callable" also includes the rocketpy' Function class.